src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -0,0 +1,695 @@
+
+                this.Wallets[wallet.Name] = walletContainer;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool DeleteWallet(string walletName)
+        {
+            WalletContainer walletContainer = this.Wallets[walletName];
+            lock (walletContainer.LockProcessBlocks)
+            {
+                lock (walletContainer.LockUpdateAccounts)
+                {
+                    lock (walletContainer.LockUpdateAddresses)
+                    {
+                        // TODO: Delete HDPayments no longer required.
+                        if (!this.DatabasePerWallet)
+                        {
+                            this.RewindWallet(walletName, null);
+
+                            int walletId = walletContainer.Wallet.WalletId;
+
+                            DBConnection conn = GetConnection(walletName);
+                            lock (conn.TransactionLock)
+                            {
+                                conn.BeginTransaction();
+                                conn.Delete<HDWallet>(walletId);
+                                conn.Execute($@"
+                                DELETE  FROM HDAddress
+                                WHERE   WalletId = {walletId}
+                                ");
+                                conn.Execute($@"
+                                DELETE  FROM HDAccount
+                                WHERE   WalletId = {walletId}
+                                ");
+                                conn.Commit();
+                            }
+
+                            return this.Wallets.TryRemove(walletName, out _);
+                        }
+                        else
+                        {
+                            DBConnection conn = GetConnection(walletName);
+                            conn.Close();
+                            File.Delete(Path.Combine(this.DBPath, $"{walletName}.db"));
+                            return this.Wallets.TryRemove(walletName, out _);
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void CreateAccount(string walletName, int accountIndex, string accountName, ExtPubKey extPubKey, DateTimeOffset? creationTime = null)
+        {
+            WalletContainer walletContainer = this.Wallets[walletName];
+
+            lock (walletContainer.LockUpdateAccounts)
+            {
+                HDWallet wallet = walletContainer.Wallet;
+
+                var conn = this.GetConnection(walletName);
+                lock (conn.TransactionLock)
+                {
+                    conn.BeginTransaction();
+
+                    var account = conn.CreateAccount(wallet.WalletId, accountIndex, accountName, extPubKey.ToString(this.Network), (int)(creationTime ?? this.DateTimeProvider.GetTimeOffset()).ToUnixTimeSeconds());
+                    conn.CreateAddresses(account, 0, 20);
```

Comment:
Maybe introducing an enum for change and non-change address types would be clearer

Reply:
Done.